### PR TITLE
[stable/fluentd-elasticsearch] several chart updates

### LINKED ISSUE #10043: [stable/fluentd-elasticsearch] change fluentd :scheme from http to https
**Version of Helm and Kubernetes**: v2.11.0 and v1.10.3

**Which chart**: [stable/fluentd-elasticsearch]

**What happened**:

I'm using the AWS Elasticsearch service, and I have installed the chart with this command:

```
helm install --name fluentd-elasticsearch --set elasticsearch.host=es.region.es.amazonaws.com,elasticsearch.port=443 stable/fluentd-elasticsearch
```

But I'm getting this error message in the logs: 

```
2018-12-17 01:12:14 +0000 [warn]: [elasticsearch] got unrecoverable error in primary and no secondary error_class=Fluent::Plugin::ElasticsearchOutput::ConnectionFailure error="Can not reach Elasticsearch cluster ({:host=>\"es.region.es.amazonaws.com\", :port=>443, :scheme=>\"http\"})!"
```

Is there a way to change the default **:scheme** from **http** to **https**?

I've also tested the connection between the pods and our Elasticsearch cluster and everything is working good.

Thanks!

@monotek 

---

**Update**: I've manually edited the Config Map to add the option: `:scheme https` and it worked!

```
2018-12-17 01:37:07 +0000 [info]: [elasticsearch] Connection opened to Elasticsearch cluster => {:host=>"es.region.es.amazonaws.com", :port=>443, :scheme=>"https"}
2018-12-17 01:37:07 +0000 [info]: [elasticsearch] Detected ES 6.x: ES 7.x will only accept `_doc` in type_name.
2018-12-17 01:37:07 +0000 [warn]: [elasticsearch] Detected ES 6.x or above and enabled insecure security: You might have to specify `ssl_version TLSv1_2` in configuration.
```

I didn't figured out yet how to make this changes permanently. 😅

Should I be worried about this warn message: `You might have to specify ssl_version TLSv1_2 in configuration`?

### PR DESCRIPTION
**Pull request by monotek**
#### What this PR does / why we need it:
- updated fluentd-elasticsearch docker image to 2.3.2
- changed configmap to be prepared for es 7.0.x
- fixed typo in README
- updated NOTES.txt & _helpers.tpl
- updated labels the helm 2.11 way
- added scheme setting

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #10043

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by monotek** [comment]
pretty please :-)

**Comment by monotek** [comment]
anyone?

File: stable/fluentd-elasticsearch/templates/_helpers.tpl [hunk 1]
```diff
@@ -9,19 +9,34 @@ Expand the name of the chart.
...
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
 
 {{/*
 Create the name of the service account to use
 */}}
 {{- define "fluentd-elasticsearch.serviceAccountName" -}}
 {{- if .Values.serviceAccount.create -}}
-    {{ default (include "fluentd-elasticsearch.fullname" .) .Values.serviceAccount.name }}
+    {{ default (include "fluentd-elasticsearch.name" .) .Values.serviceAccount.name }}
```
Comment (unguiculus) [inline stable/fluentd-elasticsearch/templates/_helpers.tpl]: This should be reverted.

File: stable/fluentd-elasticsearch/Chart.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: fluentd-elasticsearch
-version: 1.4.1
+version: 1.5.0
```
Comment (unguiculus) [inline stable/fluentd-elasticsearch/Chart.yaml]: This should be a major bump due to all the label changes.

File: stable/fluentd-elasticsearch/templates/clusterrolebinding.yaml [hunk 3]
```diff
@@ -4,12 +4,12 @@ apiVersion: rbac.authorization.k8s.io/v1
...
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
     kubernetes.io/cluster-service: "true"
     addonmanager.kubernetes.io/mode: Reconcile
-    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
-    heritage: {{ .Release.Service }}
-    release: {{ .Release.Name }}
 subjects:
 - kind: ServiceAccount
   name: {{ template "fluentd-elasticsearch.fullname" . }}
```
Comment (unguiculus) [inline stable/fluentd-elasticsearch/templates/clusterrolebinding.yaml]: `{{ template "fluentd-elasticsearch.serviceAccountName" . }}`

File: stable/fluentd-elasticsearch/templates/daemonset.yaml [hunk 4]
```diff
@@ -3,13 +3,13 @@ kind: DaemonSet
...
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
     version: {{ .Values.image.tag }}
     kubernetes.io/cluster-service: "true"
     addonmanager.kubernetes.io/mode: Reconcile
-    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
-    heritage: "{{ .Release.Service }}"
-    release: "{{ .Release.Name }}"
 spec:
   selector:
     matchLabels:
```
Comment (unguiculus) [inline stable/fluentd-elasticsearch/templates/daemonset.yaml:16]: Fix labels

**Review by unguiculus — CHANGES_REQUESTED** [review]
I just made a few comments. 

* Make sure selector labels are also updated everywhere.
* References to the service account must use `fluentd-elasticsearch.serviceAccountName` everywhere.

File: stable/fluentd-elasticsearch/templates/_helpers.tpl (hunk 1 above)
Comment (monotek) [reply stable/fluentd-elasticsearch/templates/_helpers.tpl]: done

File: stable/fluentd-elasticsearch/Chart.yaml (hunk 2 above)
Comment (monotek) [reply stable/fluentd-elasticsearch/Chart.yaml]: raised to 2.0.0

File: stable/fluentd-elasticsearch/templates/daemonset.yaml (hunk 4 above)
Comment (monotek) [reply stable/fluentd-elasticsearch/templates/daemonset.yaml:16]: done

File: stable/fluentd-elasticsearch/templates/clusterrolebinding.yaml (hunk 3 above)
Comment (monotek) [reply stable/fluentd-elasticsearch/templates/clusterrolebinding.yaml]: done

File: stable/fluentd-elasticsearch/templates/daemonset.yaml (hunk 4 above)
Comment (monotek) [reply stable/fluentd-elasticsearch/templates/daemonset.yaml:16]: done

**Comment by monotek** [comment]
@unguiculus 
Thanks for the review and some late congrats for your cncf community award :-)

**Review by unguiculus — APPROVED** [review]

**Comment by rendhalver** [comment]
It should have been noted that this change makes it impossible to upgrade to the 2.x releases from anything prior.
Removing the old labels entirely stops an upgrade from happening.
See kiwigrid/helm-charts#42 for details.

**Pull request merged**